### PR TITLE
feat: report error location when interpreting files

### PR DESCRIPTION
**Pull request by Danielmelody**

### CODE REVIEW

**Comment by Danielmelody** [comment]
Try add some invalid syntax and get the syntax error.

Semantic error location would soon been supported.

**Review by CurryPseudo — CHANGES_REQUESTED** [review]

File: src/error.rs [hunk 1]
```diff
@@ -3,22 +3,55 @@
...
+
+impl<T: PartialEq + Display> Located<T> {
+    pub fn from_data(data: T) -> Self {
+        Self {
+            data,
+            location: None,
+        }
+    }
+
+    pub fn set_location(mut self, location: Option<[u32; 2]>) -> Self {
```
[the remark this answers is not in the document]
Comment (Danielmelody) [reply src/error.rs:28]: This is a owned builder pattern

[the remark this answers is not in the document]
Comment (CurryPseudo) [reply src/error.rs:28]: Ok, I get it.
But, for builder pattern, I think with_location or located is more appropriate.

**Comment by Danielmelody** [comment]
Updated.

**Pull request merged**